src/Bedrock.Framework.Experimental/Transports/Pooling/EndPointPool.cs

```diff
@@ -0,0 +1,203 @@
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Connections;
+
+namespace Bedrock.Framework
+{
+    internal class EndPointPool : IEndPointPool, IAsyncDisposable
+    {
+        private EndPoint _endPoint;
+        private List<ConnectionContext> _idleConnections;
+        private IConnectionFactory _connectionFactory;
+        private int _maxConnectionCount;
+        private int _connectionCount;
+
+        private object SyncObj = new object();
```

Comment:
Nit: naming? Is there any standard to follow in this project?

Reply:
agree, I opened an issue about code style https://github.com/davidfowl/BedrockFramework/issues/35